src/dotnet-benchmark-client/Controllers/JobsController.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Net;
+using Benchmarks.ClientJob;
+using Microsoft.AspNetCore.Mvc;
+using Repository;
+
+namespace BenchmarkClient.Controllers
+{
+    [Route("[controller]")]
+    public class JobsController : Controller
+    {
+        private readonly IRepository<ClientJob> _jobs;
+
+        public JobsController(IRepository<ClientJob> jobs)
+        {
+            _jobs = jobs;
+        }
+
+        public IEnumerable<ClientJob> GetAll()
+        {
+            return _jobs.GetAll();
+        }
+
+        [HttpGet("{id}")]
+        public IActionResult GetById(int id)
+        {
+            var job = _jobs.Find(id);
+            if (job == null)
+            {
+                return NotFound();
+            }
+            else
+            {
+                return new ObjectResult(job);
+            }
+        }
+
+        [HttpPost]
+        public IActionResult Create([FromBody] ClientJob job)
+        {
+            if (job == null || job.Id != 0 || job.State != ClientState.Waiting)
+            {
+                return BadRequest();
+            }
+
+            job = _jobs.Add(job);
+
+            Response.Headers["Location"] = $"/jobs/{job.Id}";
+            return new StatusCodeResult((int)HttpStatusCode.Accepted);
```

Comment:
Ah.... yeah I advised you wrongly sorry. `CreatedAtAction("GetById", new { id = job.Id })` is going to interpret the anonymous object as the payload, not as route values. https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs#L887

`CreatedAtAction("GetById", null, new { id = job.Id })`  would do the trick, but it's kinda odd to pass in a null payload.

Anyways, if you feel like removing the hardcoding the URL, you should also be able to just do `Url.Action("GetById", new { id = job.Id })`. I don't really think this is important, just wanted to make mention